src/runtime/GlobalObjectBuiltinTypedArray.cpp

```diff
@@ -283,6 +282,9 @@ static Value builtinTypedArrayOf(ExecutionState& state, Value thisValue, size_t
 static Value builtinTypedArrayByteLengthGetter(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
     if (LIKELY(thisValue.isPointerValue() && thisValue.asPointerValue()->isTypedArrayObject())) {
+        if (thisValue.asObject()->asArrayBufferView()->isDetached()) {
+            return Value(0);
+        }
         return Value(thisValue.asObject()->asArrayBufferView()->byteLength());
```

Comment:
I would use a temporal ArrayBufferView pointer to hold the result of `thisValue.asObject()->asArrayBufferView()` since this pointer value is accessed twice in this method.
```c++
ArrayBufferView* buffer = thisValue.asObject()->asArrayBufferView();
```